### PR TITLE
Add config entries for Farming Station particles

### PR DESCRIPTION
**Pull request by socolio**
Nobody cares about these particles. But it can be pretty laggy with fast/large tree farms.

* decrease number of Farming Station particles (scaled with client setting)
* decrease range of Farming Station particles
* add max range of Farming Station particles to config

### CODE REVIEW

**Review by TimeConqueror — APPROVED** [review]
Looks good to me

**Comment by TimeConqueror** [comment]
> seems like it should be adressed via either a change in EIO to make the particles non laggy, or adressed in angelica. Also pls make it so the default stays the same as it was before. Config will exist thanks to this PR, so anyone will be able to reduce the particles if they like.

Why does it need t look the same as before? And as for Angelica, it would be worse to make it there I think, because some of users still use Optifine, including me

**Comment by boubou19** [comment]
Because cosmetic PRs always upsetting people, so it's preferable to opt-in the changes than to opt-out the default settings.

File: src/main/java/crazypants/util/ClientUtil.java [hunk 1]
```diff
@@ -47,18 +51,34 @@ public static void doGasLevelUpdate(int x, int y, int z, PacketGasLevel pkt) {
...
+        // 0 = All, 1 = Decreased, 2 = Minimal
+        int particleSetting = Minecraft.getMinecraft().gameSettings.particleSetting;
+        if (particleSetting >= 2 /* Minimal */) {
+            return;
+        }
+
+        int particles = particleSetting == 0 ? 6 /* All */ : 3 /* Decreased */;
+
+        World world = Minecraft.getMinecraft().theWorld;
+        EffectRenderer effectRenderer = Minecraft.getMinecraft().effectRenderer;
```
Comment (Caedis) [inline src/main/java/crazypants/util/ClientUtil.java:64]: Any reason it was changed from spawnParticle?

**Comment by boubou19** [comment]
> And as for Angelica, it would be worse to make it there I think, because some of users still use Optifine, including me

I was saying this in case the laggy code is made in base minecraft. Also optifine is officially unsupported, in favor of Angelica, so if players want to remove angelica that's fine, but done at their own risk.

File: src/main/java/crazypants/util/ClientUtil.java (hunk 1 above)
Comment (socolio) [reply src/main/java/crazypants/util/ClientUtil.java:64]: spawnParticle uses random 30% reduction for number of particles
i want more control over this number

it also contains terrible impl of particle registry (when particle name compared one by one with all possible variants)

Comment (Caedis) [reply src/main/java/crazypants/util/ClientUtil.java:64]: Good catch. Just checked it and it was a little much.
Have you tested this in game, sp and server (runServer)?

Comment (socolio) [reply src/main/java/crazypants/util/ClientUtil.java:65]: > Good catch. Just checked it and it was a little much. Have you tested this in game, sp and server (runServer)?

tested in sp and server

**Pull request merged**